### PR TITLE
extracting Data.MessagePack.RPC.

### PR DESCRIPTION
**Pull request by kazu-yamamoto**
A new module to contain the types of MessagePack RPC.

### CODE REVIEW

File: src/Data/MessagePack/RPC.hs [hunk 1]
```diff
@@ -0,0 +1,94 @@
+-- | Implementation of [MessagePack RPC](https://github.com/msgpack-rpc/msgpack-rpc/blob/master/spec.md)
+
+module Data.MessagePack.RPC (
```
Comment (igrep) [inline src/Data/MessagePack/RPC.hs]: How about renaming into `Data.MessagePack.RPC.Types`, and move `ClientViaWebSockets` into the RPC directory?
To tell the truth, I think the name `Network.MessagePack.ClientViaWebSockets` was a failure

Comment (kazu-yamamoto) [reply src/Data/MessagePack/RPC.hs]: I would like to create `Network.MessagePack.Async.Client` to contain `callRpc`, `Client`, etc.

Comment (igrep) [reply src/Data/MessagePack/RPC.hs]: Do you mean you are trying to put `callRpc` at `Async` directory though `callRpc` actually runs synchronously?
I think functions is `ClientViaWebSockets` are "asynchronous" only internally.

Comment (kazu-yamamoto) [reply src/Data/MessagePack/RPC.hs]: Yes.
See the commits below.

Comment (kazu-yamamoto) [reply src/Data/MessagePack/RPC.hs]: In the protocol level, it's async.

**Pull request closed without merging**